cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkProgramRunner.java

```diff
@@ -56,12 +58,17 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.Closeable;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
 
 /**
  * Runs {@link Spark} programs
  */
-public class SparkProgramRunner implements ProgramRunner {
+public class SparkProgramRunner extends AbstractProgramRunnerWithPlugin {
```

Comment:
not related to this PR, but should we create a task/TODO to remove ``AbstractProgramRunnerWithPlugin``, once all programs have plugins? Seems pretty redundant after every program has plugin support, perhaps ``ProgramRunner`` can become ``AbstractProgramRunner``